src/pages/index.jsx

```diff
@@ -148,14 +148,71 @@ function EpisodeSearchBar({
           d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
         ></path>
       </svg>
-      <span class="sr-only">Search</span>
     </button>
   )
 }
 
+function filterEpisodes(episodes, query) {
+  if (!query) {
+    return episodes
+  }
+  const result = []
+  const tokens = query.trim().split(/\s/)
+
+  for (const episode of episodes) {
+    if (
+      tokens
+        .some(
+          (token) =>
+            new RegExp(token, 'i').test(episode.metadata.title) ||
+            new RegExp(token, 'i').test(episode.metadata.subtitle) ||
+            new RegExp(token, 'i').test(episode.metadata.content) ||
+            new RegExp(token, 'i').test(episode.metadata.description)
+        )
+    ) {
+      result.push(episode)
+      continue
+    }
+
+    if (
+      tokens
+        .some(
+          (token) =>
+            episode.metadata.keywords &&
+            episode.metadata.keywords.split(/,\s?/).some((keyword) =>
+              new RegExp(token, 'i').test(keyword)
+            )
+        )
+    ) {
+      result.push(episode)
+      continue
+    }
+
+    if (
+      tokens
+        .some(
+          (token) =>
+            episode.metadata.author &&
+            episode.metadata.author.split(/,\s?/).some((author) =>
+              new RegExp(token, 'i').test(author)
+            )
+        )
+    ) {
+      result.push(episode)
+      continue
+    }
+  }
+
+  return result.sort(
+    ({ metadata: { pubDate: a } }, { metadata: { pubDate: b } }) =>
+      new Date(b) - new Date(a)
+  )
+}
+
 export default function Home({ episodes }) {
   const [query, setQuery] = useState('')
   const [searchClicked, setSearchClicked] = useState(false)
+  const filteredEpisodes = filterEpisodes(episodes, query)
```

Comment:
This should be memoized with `useMemo()`.

Also - would you want this to be computed on every change, or should we debounce this?
```suggestion
  const filteredEpisodes = useMemo(() => filterEpisodes(episodes, query), [episodes, query])
```